### PR TITLE
feat: support visit of stage repository

### PR DESCRIPTION
**Pull request by msgui**
## Purpose of the PR

support visit of stage repository

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [x]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [ ]  Other affects (typed here)

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [x]  `Doc - No Need`

### CODE REVIEW

**Comment by msgui** [comment]
Currently, the issue of missing the common1.0.1 package has been fixed. 
The current errors are: 
	1. JDK8 is not supported
	2.  Test phase

**Comment by imbajin** [comment]
> Currently, the issue of missing the common1.0.1 package has been fixed. The current errors are: 1. JDK8 is not supported 2. Test phase

- skip the pd/store module while we use java8?
- list the related test we could choose to skip first

File: pom.xml [hunk 1]
```diff
@@ -97,4 +97,21 @@
...
+                </repository>
+            </repositories>
+            <pluginRepositories>
+                <pluginRepository>
+                    <id>staged-releases</id>
+                    <url>https://repository.apache.org/content/groups/staging/</url>
+                </pluginRepository>
+            </pluginRepositories>
+        </profile>
+    </profiles>
```
Comment (msgui) [inline pom.xml:116]: Fix private repository package issue with 'mvn xxx -P stage
(eg: mvn clean compile -U -Dmaven.javadoc.skip=true -ntp -P stage)

**Review by msgui — COMMENTED** [review]
<profiles>
        <profile>
            <id>stage</id>
            <repositories>
                <repository>
                    <id>staged-releases</id>
                    <url>https://repository.apache.org/content/groups/staging/</url>
                </repository>
            </repositories>
            <pluginRepositories>
                <pluginRepository>
                    <id>staged-releases</id>
                    <url>https://repository.apache.org/content/groups/staging/</url>
                </pluginRepository>
            </pluginRepositories>
        </profile>
    </profiles>
Fix private repository package issue with 'mvn xxx -P stage
(eg: mvn clean compile -U -Dmaven.javadoc.skip=true -ntp **-P stage**)

**Comment by imbajin** [comment]
> > Currently, the issue of missing the common1.0.1 package has been fixed. The current errors are: 1. JDK8 is not supported 2. Test phase
> 
> - skip the pd/store module while we use java8?
> - list the related test we could choose to skip first
> 
> 

the first problem we have to handle them now

like in action add：
if java8
  mvn compile -pl hugegraph-server #specify the needed modules
else
  mvn compile # all modules

File: pom.xml [hunk 2]
```diff
@@ -94,7 +94,32 @@
...
+        <profile>
+            <id>exclude-jdk11-modules</id>
+            <activation>
+                <jdk>11</jdk>
+            </activation>
+            <modules>
+                <module>hugegraph-store</module>
+                <module>hugegraph-pd</module>
+            </modules>
+        </profile>
```
Comment (msgui) [inline pom.xml:123]: Due to the current support for JDK 11 compilation in the hugegraph-store/pd modules, it is necessary to exclude these two modules in the JDK 8.

File: pom.xml [hunk 3]
```diff
@@ -94,7 +94,32 @@
 
     <modules>
         <module>hugegraph-server</module>
-        <module>hugegraph-pd</module>
-        <module>hugegraph-store</module>
```
Comment (msgui) [inline pom.xml:98]: exclude jdk11-only modules

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -46,22 +47,31 @@ jobs:
         with:
           fetch-depth: 2
 
+      - name: copy and overwrite maven settings
```
Comment (imbajin) [inline .github/workflows/ci.yml:50]:
```suggestion
      - name: use staged maven repo settings
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -46,22 +47,31 @@ jobs:
...
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          cp  $HOME/.m2/settings.xml /tmp/settings.xml
+          mv -f .github/configs/settings.xml $HOME/.m2/settings.xml
```
Comment (imbajin) [inline .github/workflows/ci.yml:54]:
```suggestion
          cp $HOME/.m2/settings.xml /tmp/settings.xml
          mv -vf .github/configs/settings.xml $HOME/.m2/settings.xml
```

File: .github/workflows/codeql-analysis.yml [hunk 6]
```diff
@@ -33,6 +35,12 @@ jobs:
           distribution: 'zulu'
           java-version: '11'
 
+      - name: copy and overwrite maven settings
```
Comment (imbajin) [inline .github/workflows/codeql-analysis.yml:38]:
```suggestion
      - name: use staged maven repo settings
```

File: .github/workflows/codeql-analysis.yml [hunk 7]
```diff
@@ -33,6 +35,12 @@ jobs:
...
+        if: ${{ env.USE_STAGE == 'true' }}
+        run: |
+          cp  $HOME/.m2/settings.xml /tmp/settings.xml
+          mv -f .github/configs/settings.xml $HOME/.m2/settings.xml
```
Comment (imbajin) [inline .github/workflows/codeql-analysis.yml:42]:
```suggestion
          cp $HOME/.m2/settings.xml /tmp/settings.xml
          mv -vf .github/configs/settings.xml $HOME/.m2/settings.xml
```

**Pull request merged**